src/parsers/perf/perfparser.cpp

```diff
@@ -1351,46 +1395,71 @@ void PerfParser::startParseFile(const QString& path, const QString& sysroot, con
         return;
     }
 
-    QStringList parserArgs = {QStringLiteral("--input"), decompressIfNeeded(path), QStringLiteral("--max-frames"),
-                              QStringLiteral("1024")};
-    if (!sysroot.isEmpty()) {
-        parserArgs += {QStringLiteral("--sysroot"), sysroot};
-    }
-    if (!kallsyms.isEmpty()) {
-        parserArgs += {QStringLiteral("--kallsyms"), kallsyms};
-    }
-    if (!debugPaths.isEmpty()) {
-        parserArgs += {QStringLiteral("--debug"), debugPaths};
-    }
-    if (!extraLibPaths.isEmpty()) {
-        parserArgs += {QStringLiteral("--extra"), extraLibPaths};
-    }
-    if (!appPath.isEmpty()) {
-        parserArgs += {QStringLiteral("--app"), appPath};
-    }
-    if (!arch.isEmpty()) {
-        parserArgs += {QStringLiteral("--arch"), arch};
-    }
+    auto parserArgs = [this](const QString& filename){
```

Comment:
this hunk here should be a separate cleanup patch which we can merge earlier then to reduce the size of this diff patch